### ISSUE TITLE
Extension not found

### ISSUE THREAD

**Issue by DMEbner** [rongrimes/weewx-tempest-pi#1]
I downloaded the file into my Pi download folder and ran this command and it says it can't find this extension:

$ sudo wee_extension --install weatherflow-udp-master.zip

Can anyone give me a little help? :)

pi@raspberrypi:~ $ sudo wee_extension --install weatherflow-udp-master.zip
Request to install 'weatherflow-udp-master.zip'
Traceback (most recent call last):
  File "/usr/share/weewx/wee_extension", line 92, in <module>
    main()
  File "/usr/share/weewx/wee_extension", line 84, in main
    ext.install_extension(options.install)
  File "/usr/share/weewx/weecfg/extension.py", line 139, in install_extension
    raise InstallError("Extension '%s' not found." % extension_path)
weecfg.extension.InstallError: Extension 'weatherflow-udp-master.zip' not found.
pi@raspberrypi:~ $

**Comment by rongrimes** [comment]
I just ran the installation following my document on a Pi 4 and installed weatherflow-udp-master.zip successfully so `weewx` and PiOS are current versions. I'm using:
```
pi@raspberrypi5:~ $ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
```
You say you downloaded the zip file into your `Download` folder. I loaded the file into my home directory (/home/pi), and ran `wee_extension` from my home directory. Maybe you just need to move the file into the home directory.

Let me know if you succeed.

**Comment by DMEbner** [comment]
This did indeed work - after moving to the home directory, it ran without issue. Thanks, again!

**Comment by DMEbner** [comment]
Any reason you can see that my data is not filtering out to WeeWx and the map?



'##############################################################################

# This section is for general configuration information.

# Set to 1 for extra debug info, otherwise comment it out or set to zero
debug = 1

# Root directory of the weewx data file hierarchy for this station
WEEWX_ROOT = /

# Whether to log successful operations. May get overridden below.
log_success = True

# Whether to log unsuccessful operations. May get overridden below.
log_failure = True

# Do not modify this. It is used when installing and updating weewx.
version = 4.10.2

##############################################################################

#   This section is for information about the station.

[Station]
    
    # Description of the station location
    location = "Hallsville, MO"
    
    # Latitude in decimal degrees. Negative for southern hemisphere
    latitude = 39.06260
    # Longitude in decimal degrees. Negative for western hemisphere.
    longitude = -92.258197
    
    # Altitude of the station, with the unit it is in. This is used only
    # if the hardware cannot supply a value.
    altitude = 922, feet 
    
    # Set to type of station hardware. There must be a corresponding stanza
    # in this file, which includes a value for the 'driver' option.
    station_type = WeatherFlowUDP
    
    # If you have a website, you may specify an URL. This is required if you
    # intend to register your station.
    station_url = https://tempestwx.com/station/43888/
    
    # The start of the rain year (1=January; 10=October, etc.). This is
    # downloaded from the station if the hardware supports it.
    rain_year_start = 1
    
    # Start of week (0=Monday, 6=Sunday)
    week_start = 6

##############################################################################

[WeatherFlowUDP]
    driver = user.weatherflowudp
    log_raw_packets = False
    #   log_raw_packets = True
    #   udp_address = <broadcast>
    #   udp_address = 0.0.0.0
    udp_address = 255.255.255.255
    udp_port = 50222
    udp_timeout = 90
    share_socket = False

     # This section is for the TEMPEST WeatherFlow Bridge packets, via UDP broadcast on local subnet

    [[sensor_map]]
        outTemp = air_temperature.ST-00086816.obs_st
        outHumidity = relative_humidity.ST-00086816.obs_st
        pressure = station_pressure.ST-00086816.obs_st
        lightning_strikes =  lightning_strike_count.ST-00086816.obs_st
        avg_distance =  lightning_strike_avg_distance.ST-00086816.obs_st
        outTempBatteryStatus = battery.ST-00086816.obs_st
        windSpeed = wind_speed.ST-00086816.rapid_wind
        windDir = wind_direction.ST-00086816.rapid_wind
        luxXXX = illuminance.ST-00086816.obs_st
        UV = uv.ST-00086816.obs_st
        rain = rain_accumulated.ST-00086816.obs_st
        windBatteryStatus = battery.ST-00086816.obs_st
        radiation = solar_radiation.ST-00086816.obs_st
        lightningXXX = distance.ST-00086816.evt_strike
        lightningYYY = energy.ST-00086816.evt_strike


##############################################################################

#   This section is for uploading data to Internet sites

[StdRESTful]
    
    # Uncomment and change to override logging for uploading services.
    # log_success = True
    # log_failure = True
    
    [[StationRegistry]]
        # To register this weather station with weewx, set this to true,
        # then fill out option 'station_url', located in the [Station] section above.
        register_this_station = true 
    
    [[AWEKAS]]
        # This section is for configuring posts to AWEKAS.
        
        # If you wish to post to AWEKAS, set the option 'enable' to true, then specify a username
        # and password. To guard against parsing errors, put the password in quotes.
        enable = false
        username = replace_me
        password = replace_me
    
    [[CWOP]]
        # This section is for configuring posts to CWOP.
        
        # If you wish to post to CWOP, set the option 'enable' to true,
        # then specify the station ID (e.g., CW1234).
        enable = false
        station = replace_me
        # If this is an APRS (radio amateur) station, specify the
        # passcode (e.g., 12345). Otherwise, ignore.
        passcode = replace_me
    
    
    [[PWSweather]]
        # This section is for configuring posts to PWSweather.com.
        
        # If you wish to post to PWSweather.com, set the option 'enable' to true, then specify a
        # station and password. To guard against parsing errors, put the password in quotes.
        enable = false
        station = replace_me
        password = replace_me
    
    [[WOW]]
        # This section is for configuring posts to WOW.
        
        # If you wish to post to WOW, set the option 'enable' to true, then specify a station and
        # password. To guard against parsing errors, put the password in quotes.
        enable = false
        station = replace_me
        password = replace_me
    
    [[Wunderground]]
        # This section is for configuring posts to the Weather Underground.
        
        # If you wish to post to the Weather Underground, set the option 'enable' to true,  then
        # specify a station (e.g., 'KORHOODR3') and password. To guard against parsing errors, put
        # the password in quotes.
        enable = false
        station = replace_me
        password = replace_me
        
        # If you plan on using wunderfixer, set the following
        # to your API key:
        api_key = replace_me
        
        # Set the following to True to have weewx use the WU "Rapidfire"
        # protocol. Not all hardware can support it. See the User's Guide.
        rapidfire = False

##############################################################################

#   This section specifies what reports, using which skins, to generate.

[StdReport]
    
    # Where the skins reside, relative to WEEWX_ROOT
    SKIN_ROOT = /etc/weewx/skins
    
    # Where the generated reports should go, relative to WEEWX_ROOT
    HTML_ROOT = /var/www/html/weewx
    
    # Uncomment and change to override logging for reports
    # log_success = True
    # log_failure = True
    
    # The database binding indicates which data should be used in reports.
    data_binding = wx_binding
    
    # Each of the following subsections defines a report that will be run.
    # See the customizing guide to change the units, plot types and line
    # colors, modify the fonts, display additional sensor data, and other
    # customizations. Many of those changes can be made here by overriding
    # parameters, or by modifying templates within the skin itself.
    
    [[SeasonsReport]]
        # The SeasonsReport uses the 'Seasons' skin, which contains the
        # images, templates and plots for the report.
        skin = Seasons
        enable = true
    
    [[SmartphoneReport]]
        # The SmartphoneReport uses the 'Smartphone' skin, and the images and
        # files are placed in a dedicated subdirectory.
        skin = Smartphone
        enable = false
        HTML_ROOT = /var/www/html/weewx/smartphone
    
    [[MobileReport]]
        # The MobileReport uses the 'Mobile' skin, and the images and files
        # are placed in a dedicated subdirectory.
        skin = Mobile
        enable = false
        HTML_ROOT = /var/www/html/weewx/mobile
    
    [[StandardReport]]
        # This is the old "Standard" skin. By default, it is not enabled.
        skin = Standard
        enable = false
    
    [[FTP]]
        # FTP'ing the results to a webserver is treated as just another report,
        # albeit one with an unusual report generator!
        skin = Ftp
        
        # If you wish to use FTP, set "enable" to "true", then
        # fill out the next four lines.
        # Use quotes around passwords to guard against parsing errors.
        enable = false
        user = replace_me
        password = replace_me
        server = replace_me    # The ftp server name, e.g, www.myserver.org
        path = replace_me    # The destination directory, e.g., /weather
        
        # Set to True for an FTP over TLS (FTPS) connection. Not all servers
        # support this.
        secure_ftp = False
        
        # To upload files from something other than what HTML_ROOT is set
        # to above, specify a different HTML_ROOT here.
        #HTML_ROOT = /var/www/html/weewx
        
        # Most FTP servers use port 21
        port = 21
        
        # Set to 1 to use passive mode, zero for active mode
        passive = 1
    `

**Comment by rongrimes** [comment]
Hi DME:
Differences with my .conf are:

```
[Station]
    # Description of the station location
    location = Hollywood Cres Toronto Canada
```


(no quotes).
Also:

```
    [[StationRegistry]]
        # To register this weather station with weewx, set this to true,
        # then fill out option 'station_url' above.
        register_this_station = True
```

(**True** in place of **true**, but I don't really think this makes a difference.)

I see that in my `/etc/weewx` directory, I have an old copy of a .conf file, and it has the location name in quotes. I seem to remember having the same trouble in getting my station recognized and dropping the quotes helped (it was nearly 2 years ago and that brain cell has since died). Note that you'll need to drop the comma in your _location_ name.

Please let me know if this works.

**Comment by DMEbner** [comment]
Good morning!

 Unfortunately I am still not on the Wee map with the corrections you suggested. I did restart WeeWX after making updates. 

 When I look at /var/www/….. etc there is weather data there but it is not mine and hasn't been about 70 minutes since it updated. I did find/replace all with my Tempest serial number.

Any other suggestions? I can resend you my conf if that would help. 

I really appreciate the help.

Cheers!

-Dan
